tests/filewriter_test.cpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file filewriter_test.cpp
+ * @author Jeffin Sam
+ */
+#include "catch.hpp"
+#include "mlboard.hpp"
+#include <unistd.h>
+#include <sstream>
+#include <cstdio>
+#include <sys/stat.h>
+/**
+ * Test two Filewriter objects created at the same time.
+ */
+TEST_CASE("Writing two files at a time", "[FileWriter]")
+{
+    // Both files created at same time and hence there should be only one file.
+    mlboard::FileWriter f1,f2;
+    mlboard::SummaryWriter<mlboard::FileWriter>::Scalar("Sample_1",1,1.1,f1);
+    mlboard::SummaryWriter<mlboard::FileWriter>::Scalar("Sample_2",1,1.1,f2);
+    REQUIRE(f1.FileName() == f2.FileName());
+    remove(f1.FileName().c_str());
+}
+
+/**
+ * Test the 2 filewriter objects at a different dir time.
+ */
+TEST_CASE("Writing two files at a time in different paths", "[FileWriter]")
+{
+    // Create temp dirs.
+    mkdir("_temp1_",0777);
+    mkdir("_temp2_",0777);
+
+    mlboard::FileWriter f1("_temp1_"),f2("_temp2_");
+    mlboard::SummaryWriter<mlboard::FileWriter>::Scalar("Sample_1",1,1.1,f1);
+    mlboard::SummaryWriter<mlboard::FileWriter>::Scalar("Sample_2",1,1.1,f2);
+    REQUIRE(f1.FileName() != f2.FileName());
+
+    REQUIRE(f1.FileName().substr(8,f1.FileName().length()) ==
+        f2.FileName().substr(8,f2.FileName().length()));
+
+    // Remove temp dirs.
+    remove(f1.FileName().c_str());
+    remove(f2.FileName().c_str());
+    rmdir("_temp1_");
```

Comment:
This is not removing "_temp1_"

Reply:
I am not sure what does that mean, could you clarify it

Reply:
the directory "_temp1_" is not getting removed. with this command. After running ./mlboard_tests a blank directories "_tem1_" and "_temp2_" exist.

Reply:
Can you try with fresh build dir ??? 
I think may be it's some old files 

I am just creating _temp 

Because there could be chances that build has temp 
And thus differentiating it from user name convention 


I would be creating _temp folders

Reply:
_ is considered as wrapper for italics in markdown. That's why it got italic style. I intended to say "\_temp1\_" and "\_temp2\_" folders they exist after running test

Reply:
May be since yours is ubuntu that's why !!!

Reply:
I am slightly worried that whether Unix functions (remove, rmdir etc.) would work on windows. So looking for a solution. Let's wait for two days else we will merge it.

Reply:
It looks like you've got boost -- https://www.boost.org/doc/libs/1_45_0/libs/filesystem/v3/doc/reference.html#remove_all
(Alternatively, there's an equivalent in std starting c++17)

Reply:
Yeah, I was knowing about `boost:: filesystem`, Let's not add boost as a dependency.  
Currently, To solve the issue which I was facing I am removing the directories after catch session run. 
The code which I have pushed doesn't solve the windows issue. I am having not so great solution for windows but to test that I will need to setup mlboard on windows first. It would take some time so let's resolve the windows problem in a separate PR . I will make a PR to resolve the problem soon.
@jeffin143 @birm let me know what you think.

Reply:
Yeah I also don't prefer boost , your solution seems ok to me

Let's debug windows later after setting up Mlpack in windows , probably in a different pr